atlasdb-jdbc/src/main/java/com/palantir/atlasdb/keyvalue/jdbc/JdbcKeyValueService.java

```diff
@@ -233,13 +234,14 @@ public Void apply(DSLContext ctx) {
         if (Iterables.isEmpty(rows)) {
             return ImmutableMap.of();
         }
-        return run(new Function<DSLContext, Map<Cell, Value>>() {
-            @Override
-            public Map<Cell, Value> apply(DSLContext ctx) {
+
+        Map<Cell, Value> toReturn = new HashMap<>();
+        for (List<byte[]> part : Iterables.partition(rows, PARTITION_SIZE)) {
```

Comment:
A little confused here - it seems this is called through `getRows()`, which already guarantees that the iterable passed in to this method already has at most `PARTITION_SIZE` elements.

Do we intend `batchSizeForReads` here? If this is intended to read `PARTITION_SIZE` then we shouldn't need this partition at all, and can just run the single `toReturn.putAll(run(...))` call once.